### PR TITLE
call gql function passed down for funding eligibility

### PR DESCRIPTION
**Pull request by vishakha94**
* receives callgqlforfundingeligibility handler
* retrieves funding eligibility on its own and not depend on req.query

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "scripts": {
     "start": "babel-node ./server/server",
     "setup": "npm install && npm run flow-typed",
-    "lint": "eslint src/ server/ test/ *.js",
+    "lint": "eslint --fix src/ server/ test/ *.js",
```
Comment (bluepnume) [inline package.json]: Let's not change this

File: server/button.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { serverErrorResponse, clientErrorResponse, htmlResponse, allowFrame, def
 import type { ExpressRequest, ExpressResponse, LoggerType } from './types';
 import { buttonStyle } from './style';
 
-export function getButtonMiddleware({ logger = defaultLogger } : { logger? : LoggerType } = {}) : (req : ExpressRequest, res : ExpressResponse) => Promise<void> {
+export function getButtonMiddleware({ logger = defaultLogger, callGqlForFundingEligibility } : { logger? : LoggerType, callGqlForFundingEligibility : Function }) : (req : ExpressRequest, res : ExpressResponse) => Promise<void> {
```
Comment (bluepnume) [inline server/button.js]: Can we go with `getFundingEligibility`? We don't really care at this point whether it's graphql or something else.

File: server/button.js [hunk 3]
```diff
@@ -29,7 +29,15 @@ export function getButtonMiddleware({ logger = defaultLogger } : { logger? : Log
             logger.info(req, `button_render_version_${ render.version }`);
 
             const params = undotify(req.query);
-            const { clientID, fundingEligibility, nonce } = getParams(params, req, res);
+            const receivedParams = getParams(params, req, res);
+            const { clientID, currency, intent, commit, vault, disableFunding, disableCard, merchantID, buttonSessionID, nonce } = receivedParams;
+            // default to pp geo loc if not passed through sdk
+            const { buyerCountry = req.get(HTTP_HEADER.PP_GEO_LOC) } =  receivedParams;
```
Comment (bluepnume) [inline server/button.js]: Can we just destructure everything at once?

File: server/button.js [hunk 4]
```diff
@@ -29,7 +29,15 @@ export function getButtonMiddleware({ logger = defaultLogger } : { logger? : Log
...
+            const gqlArg = { clientId:        clientID, buyerCountry, cookies:         req.get('cookie'), ip:              req.ip,
+                currency, intent, commit, vault, disableFunding, disableCard, merchantId:      merchantID,
```
Comment (bluepnume) [inline server/button.js]: Weird formatting :D

File: package.json (hunk 1 above)
Comment (vishakha94) [reply package.json]: sorry! I always add that temporarily. missed to remove it.

File: server/button.js (hunk 4 above)
Comment (vishakha94) [reply server/button.js]: blame eslint! --fix

**Comment by bluepnume** [comment]
Looks good! will merge once we've got the smartcomponentnodeweb changes in

File: server/button.js [hunk 5]
```diff
@@ -29,7 +29,36 @@ export function getButtonMiddleware({ logger = defaultLogger } : { logger? : Log
...
-            const { clientID, fundingEligibility, nonce } = getParams(params, req, res);
+            const receivedParams = getParams(params, req, res);
+            // default buyerCountry to pp geo loc if not passed through sdk
+            const { clientID, currency, intent, commit, vault, buyerCountry = req.get(HTTP_HEADER.PP_GEO_LOC), disableFunding, disableCard, merchantID, buttonSessionID, clientAccessToken, nonce } = receivedParams;
+            const gqlArg = { clientId: clientID, buyerCountry, cookies: req.get('cookie'), ip: req.ip, currency, intent, commit, vault, disableFunding, disableCard, merchantId: merchantID, userAgent: req.headers['user-agent'], buttonSessionId: buttonSessionID, clientAccessToken };
+            let fundingEligibility;
+            try {
+                fundingEligibility = await getFundingEligibility(req, gqlArg);
+            } catch (err) {
+                logger.warn(req, 'gql_errored_for_fundingEligibility', { err: err.stack ? err.stack : err.toString() });
```
Comment (bluepnume) [inline server/button.js]: Let's do logger.error. This is a serious one :D

Comment (vishakha94) [inline server/button.js]:
```suggestion
                logger.error(req, 'gql_errored_for_fundingEligibility', { err: err.stack ? err.stack : err.toString() });
```

**Pull request merged**